src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ClusterApplierServiceStatsCollector.java

```diff
@@ -0,0 +1,164 @@
+            MetricsConfiguration.CONFIG_MAP.get(ClusterApplierServiceStatsCollector.class).samplingInterval;
+    private static final int KEYS_PATH_LENGTH = 0;
+    private static final Logger LOG = LogManager.getLogger(ClusterApplierServiceStatsCollector.class);
+    private static final String GET_CLUSTER_APPLIER_SERVICE_STATS_METHOD_NAME = "getStats";
+    private static final ObjectMapper mapper = new ObjectMapper();
+    private final StringBuilder value;
+    private final PerformanceAnalyzerController controller;
+    private final ConfigOverridesWrapper configOverridesWrapper;
+
+    private MetricsTracker tracker;
+
+    public ClusterApplierServiceStatsCollector(PerformanceAnalyzerController controller,
+                                               ConfigOverridesWrapper configOverridesWrapper) {
+        super(SAMPLING_TIME_INTERVAL, ClusterApplierServiceStatsCollector.class.getSimpleName());
+        value = new StringBuilder();
+        this.controller = controller;
+        this.configOverridesWrapper = configOverridesWrapper;
+        this.tracker = new MetricsTracker();
+    }
+
+    @Override
+    public void collectMetrics(long startTime) {
+        if(!controller.isCollectorEnabled(configOverridesWrapper, getCollectorName())) {
+            return;
+        }
+        try {
+            long mCurrT = System.currentTimeMillis();
+            if (ESResources.INSTANCE.getClusterService() == null
+                    || ESResources.INSTANCE.getClusterService().getClusterApplierService() == null) {
+                return;
+            }
+            mapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
+            mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+            ClusterApplierServiceStats clusterApplierServiceStats = null;
+            try {
+                clusterApplierServiceStats = mapper.readValue(
+                        mapper.writeValueAsString(getClusterApplierServiceStats()), ClusterApplierServiceStats.class);
+            } catch (InvocationTargetException | IllegalAccessException | NoSuchMethodException ex) {
+                LOG.warn("No method found to get cluster state applier thread stats. " +
+                        "Skipping ClusterApplierServiceStatsCollector");
+                return;
+            }
+
+            ClusterApplierServiceMetrics clusterApplierServiceMetrics = new ClusterApplierServiceMetrics(
+                    computeLatency(clusterApplierServiceStats), computeFailure(clusterApplierServiceStats));
+
+            value.setLength(0);
+            value.append(PerformanceAnalyzerMetrics.getJsonCurrentMilliSeconds())
+                    .append(PerformanceAnalyzerMetrics.sMetricNewLineDelimitor);
+            value.append(clusterApplierServiceMetrics.serialize());
+            saveMetricValues(value.toString(), startTime);
+            this.tracker = new MetricsTracker(clusterApplierServiceStats.timeTakenInMillis,
```

Comment:
the scope of this tracker object is class level but you are initializing it again in collectMetrics method. Multiple collectMetrics() call could result in race conditions. I guess it is making the assumption that applier thread is single threaded. It would be better not to keep it at class level and pass it as parameters in the below methods.